src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Oid.cs

```diff
@@ -53,8 +55,27 @@ public static partial class AsnDecoder
             out int bytesConsumed,
             Asn1Tag? expectedTag = null)
         {
-            // TODO: Inline this call when it won't cause a PR/diff problem.
-            return ReadObjectIdentifier(source, ruleSet, expectedTag, out bytesConsumed);
+            // T-REC-X.690-201508 sec 8.19.1
+            ReadOnlySpan<byte> contents = GetPrimitiveContentSpan(
+                source,
+                ruleSet,
+                expectedTag ?? Asn1Tag.ObjectIdentifier,
+                UniversalTagNumber.ObjectIdentifier,
+                out int consumed);
+
+#if NETCOREAPP
+            string? wellKnown = WellKnownOids.GetValue(contents);
+
+            if (wellKnown is not null)
+            {
+                bytesConsumed = consumed;
+                return wellKnown;
+            }
+#endif
+
+            string ret = ReadObjectIdentifier(contents);
```

Comment:
Probably as a follow up, but since most dotted-decimal OIDs fit in a `stackalloc char[256]`, maybe this path could use `ValueStringBuilder` instead of `StringBuilder`.